spec/puppet-lint/plugins/check_classes/arrow_on_right_operand_line_spec.rb

```diff
@@ -11,7 +11,8 @@
 
       context 'arrow on the line of left operand' do
         let(:code) do
-          "Package['httpd']  #{operator}
+          "
+            Package['httpd']  #{operator}
```

Comment:
To clarify - I think that both the previous version and the current version of this test could be kept. Because your changes includes handling of optional newline and whitespace tokens; it may be valuable to test both that:

a. arrow_on_right_operand_line.rb's code works when line and space characters appear
b. arrow_on_right_operand_line.rb's code does not error when there are no line or space tokens